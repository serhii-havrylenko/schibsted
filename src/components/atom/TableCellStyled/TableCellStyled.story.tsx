import { Table, TableBody, TableHead, TableRow } from '@material-ui/core';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import TableCellStyled from './TableCellStyled';

storiesOf('TableCellStyled', module).add('default', () => (
  <Table>
    <TableHead>
      {' '}
      <TableRow>
        <TableCellStyled>Head cell</TableCellStyled>
      </TableRow>
    </TableHead>
    <TableBody>
      <TableRow>
        <TableCellStyled>Body cell</TableCellStyled>
      </TableRow>
    </TableBody>
  </Table>
));
