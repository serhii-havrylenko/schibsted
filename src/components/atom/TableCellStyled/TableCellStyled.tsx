import { withStyles } from '@material-ui/core';
import TableCell from '@material-ui/core/TableCell';

export const TableCellStyled = withStyles(() => ({
  head: {
    backgroundColor: '#65a1ec',
    color: '#fff',
  },
  body: {
    fontSize: 14,
    whiteSpace: 'pre',
    '&:first-child': {
      whiteSpace: 'nowrap',
    },
  },
}))(TableCell);

export default TableCellStyled;
