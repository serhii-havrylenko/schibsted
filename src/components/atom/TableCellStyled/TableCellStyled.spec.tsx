import { Table, TableBody, TableHead, TableRow } from '@material-ui/core';
import { mount } from 'enzyme';
import * as React from 'react';

import TableCellStyled from './TableCellStyled';

describe('<TableCellStyled /> tests', () => {
  test('should match snapshot for head cell', () => {
    const tree = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableCellStyled>Cell</TableCellStyled>
          </TableRow>
        </TableHead>
      </Table>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot for body cell', () => {
    const tree = mount(
      <Table>
        <TableBody>
          <TableRow>
            <TableCellStyled>Cell</TableCellStyled>
          </TableRow>
        </TableBody>
      </Table>,
    );
    expect(tree).toMatchSnapshot();
  });
});
