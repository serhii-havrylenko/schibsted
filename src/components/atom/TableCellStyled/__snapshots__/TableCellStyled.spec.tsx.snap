// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCellStyled /> tests should match snapshot for body cell 1`] = `
<WithStyles(Table)>
  <Table
    classes={
      Object {
        "root": "MuiTable-root-1",
      }
    }
    component="table"
    padding="default"
  >
    <table
      className="MuiTable-root-1"
    >
      <WithStyles(TableBody)>
        <TableBody
          classes={
            Object {
              "root": "MuiTableBody-root-22",
            }
          }
          component="tbody"
        >
          <tbody
            className="MuiTableBody-root-22"
          >
            <WithStyles(TableRow)>
              <TableRow
                classes={
                  Object {
                    "footer": "MuiTableRow-footer-7",
                    "head": "MuiTableRow-head-6",
                    "hover": "MuiTableRow-hover-5",
                    "root": "MuiTableRow-root-3",
                    "selected": "MuiTableRow-selected-4",
                  }
                }
                component="tr"
                hover={false}
                selected={false}
              >
                <tr
                  className="MuiTableRow-root-3"
                >
                  <WithStyles(WithStyles(TableCell))>
                    <WithStyles(TableCell)
                      classes={
                        Object {
                          "body": "WithStyles-TableCell--body-9",
                          "head": "WithStyles-TableCell--head-8",
                        }
                      }
                    >
                      <TableCell
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTableCell-alignCenter-19",
                            "alignJustify": "MuiTableCell-alignJustify-21",
                            "alignLeft": "MuiTableCell-alignLeft-18",
                            "alignRight": "MuiTableCell-alignRight-20",
                            "body": "MuiTableCell-body-12 WithStyles-TableCell--body-9",
                            "footer": "MuiTableCell-footer-13",
                            "head": "MuiTableCell-head-11 WithStyles-TableCell--head-8",
                            "numeric": "MuiTableCell-numeric-14",
                            "paddingCheckbox": "MuiTableCell-paddingCheckbox-16",
                            "paddingDense": "MuiTableCell-paddingDense-15",
                            "paddingNone": "MuiTableCell-paddingNone-17",
                            "root": "MuiTableCell-root-10",
                          }
                        }
                      >
                        <td
                          aria-sort={null}
                          className="MuiTableCell-root-10 MuiTableCell-body-12 WithStyles-TableCell--body-9"
                        >
                          Cell
                        </td>
                      </TableCell>
                    </WithStyles(TableCell)>
                  </WithStyles(WithStyles(TableCell))>
                </tr>
              </TableRow>
            </WithStyles(TableRow)>
          </tbody>
        </TableBody>
      </WithStyles(TableBody)>
    </table>
  </Table>
</WithStyles(Table)>
`;

exports[`<TableCellStyled /> tests should match snapshot for head cell 1`] = `
<WithStyles(Table)>
  <Table
    classes={
      Object {
        "root": "MuiTable-root-1",
      }
    }
    component="table"
    padding="default"
  >
    <table
      className="MuiTable-root-1"
    >
      <WithStyles(TableHead)>
        <TableHead
          classes={
            Object {
              "root": "MuiTableHead-root-2",
            }
          }
          component="thead"
        >
          <thead
            className="MuiTableHead-root-2"
          >
            <WithStyles(TableRow)>
              <TableRow
                classes={
                  Object {
                    "footer": "MuiTableRow-footer-7",
                    "head": "MuiTableRow-head-6",
                    "hover": "MuiTableRow-hover-5",
                    "root": "MuiTableRow-root-3",
                    "selected": "MuiTableRow-selected-4",
                  }
                }
                component="tr"
                hover={false}
                selected={false}
              >
                <tr
                  className="MuiTableRow-root-3 MuiTableRow-head-6"
                >
                  <WithStyles(WithStyles(TableCell))>
                    <WithStyles(TableCell)
                      classes={
                        Object {
                          "body": "WithStyles-TableCell--body-9",
                          "head": "WithStyles-TableCell--head-8",
                        }
                      }
                    >
                      <TableCell
                        align="inherit"
                        classes={
                          Object {
                            "alignCenter": "MuiTableCell-alignCenter-19",
                            "alignJustify": "MuiTableCell-alignJustify-21",
                            "alignLeft": "MuiTableCell-alignLeft-18",
                            "alignRight": "MuiTableCell-alignRight-20",
                            "body": "MuiTableCell-body-12 WithStyles-TableCell--body-9",
                            "footer": "MuiTableCell-footer-13",
                            "head": "MuiTableCell-head-11 WithStyles-TableCell--head-8",
                            "numeric": "MuiTableCell-numeric-14",
                            "paddingCheckbox": "MuiTableCell-paddingCheckbox-16",
                            "paddingDense": "MuiTableCell-paddingDense-15",
                            "paddingNone": "MuiTableCell-paddingNone-17",
                            "root": "MuiTableCell-root-10",
                          }
                        }
                      >
                        <th
                          aria-sort={null}
                          className="MuiTableCell-root-10 MuiTableCell-head-11 WithStyles-TableCell--head-8"
                          scope="col"
                        >
                          Cell
                        </th>
                      </TableCell>
                    </WithStyles(TableCell)>
                  </WithStyles(WithStyles(TableCell))>
                </tr>
              </TableRow>
            </WithStyles(TableRow)>
          </thead>
        </TableHead>
      </WithStyles(TableHead)>
    </table>
  </Table>
</WithStyles(Table)>
`;
